dotnet/src/SemanticKernel.Core/Planning/PlannerConfigBase.cs

```diff
@@ -15,52 +15,65 @@ namespace Microsoft.SemanticKernel.Planning;
+        /// </summary>
+        /// <remarks>
+        /// Limits the number of relevant functions as result of semantic
+        /// search included in the plan creation request.
+        /// <see cref="IncludedFunctions"/> will be included
+        /// in the plan regardless of this limit.
+        /// </remarks>
+        public int MaxRelevantFunctions { get; set; } = 100;
+
+        /// <summary>
+        /// The minimum relevancy score for a function to be considered.
+        /// </summary>
+        /// <remarks>
+        /// Depending on the embeddings engine used, the user ask, the step goal
+        /// and the functions available, this value may need to be adjusted.
+        /// For default, this is set to null which will return the top
+        /// <see cref="MaxRelevantFunctions"/> sorted by relevancy.
+        /// </remarks>
+        public double? RelevancyThreshold { get; set; }
+    }
 
     /// <summary>
-    /// When using <see cref="Memory"/> to get relevant functions,
-    /// this list of functions will be included regardless of relevancy.
+    /// Delegate to get the prompt template string.
     /// </summary>
-    public HashSet<(string, string)> IncludedFunctions { get; } = new();
+    public Func<string>? GetPromptTemplate { get; set; } = null;
 
     /// <summary>
-    /// Semantic memory to use for function lookup (optional).
-    /// This property will be ignored if <see cref="GetAvailableFunctionsAsync"/> is set.
+    /// A list of plugins to exclude from the plan creation request.
     /// </summary>
-    public ISemanticTextMemory Memory { get; set; } = NullMemory.Instance;
+    public HashSet<string> ExcludedPlugins { get; } = new();
 
     /// <summary>
-    /// The maximum number of relevant functions to include in the plan.
+    /// A list of functions to exclude from the plan creation request.
     /// </summary>
-    /// <remarks>
-    /// Limits the number of relevant functions as result of semantic
-    /// search included in the plan creation request.
-    /// <see cref="IncludedFunctions"/> will be included
-    /// in the plan regardless of this limit.
-    /// </remarks>
-    public int MaxRelevantFunctions { get; set; } = 100;
+    public HashSet<string> ExcludedFunctions { get; } = new();
 
     /// <summary>
-    /// The minimum relevancy score for a function to be considered
+    /// Semantic Memory configuration, used to enable function filtering during plan creation.
     /// </summary>
     /// <remarks>
-    /// Depending on the embeddings engine used, the user ask, the step goal
-    /// and the functions available, this value may need to be adjusted.
-    /// For default, this is set to null to exhibit previous behavior.
+    /// This configuration will be ignored if <see cref="GetAvailableFunctionsAsync"/> is set.
     /// </remarks>
-    public double? RelevancyThreshold { get; set; }
+    public SemanticMemoryConfig SemanticMemory { get; set; } = new();
```

Comment:
nit: I prefer to be more specific about what this is so it isn't confused with the `ISemanticTextMemory `
```suggestion
    public SemanticMemoryConfig SemanticMemoryConfig { get; set; } = new();
```

Reply:
Going with `SemanticMemoryConfiguration`